pkg/webui/console/views/gateway-add/index.js

```diff
@@ -0,0 +1,182 @@
+import { push } from 'connected-react-router'
+
+import Breadcrumb from '../../../components/breadcrumbs/breadcrumb'
+import { withBreadcrumb } from '../../../components/breadcrumbs/context'
+import sharedMessages from '../../../lib/shared-messages'
+import Message from '../../../lib/components/message'
+import Form from '../../../components/form'
+import Field from '../../../components/field'
+import SubmitBar from '../../../components/submit-bar'
+import Button from '../../../components/button'
+import { id as gatewayIdRegexp } from '../../lib/regexp'
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import FrequencyPlansSelect from '../../containers/freq-plans-select'
+
+import api from '../../api'
+
+import { userIdSelector } from '../../store/selectors/user'
+
+import style from './gateway-add.styl'
+
+const m = defineMessages({
+  dutyCycle: 'Enforce Duty Cycle',
+  gatewayIdPlaceholder: 'my-new-gateway',
+  createGateway: 'Create Gateway',
+})
+
+const initialValues = {
+  enforce_duty_cycle: true,
+}
+
+const validationSchema = Yup.object().shape({
+  ids: Yup.object().shape({
+    gateway_id: Yup.string()
+      .matches(gatewayIdRegexp, sharedMessages.validateAlphanum)
+      .min(2, sharedMessages.validateTooShort)
+      .max(36, sharedMessages.validateTooLong)
+      .required(sharedMessages.validateRequired),
+    eui: Yup.string()
+      .length(8 * 2, sharedMessages.validateTooShort),
+  }),
+  name: Yup.string()
+    .min(2, sharedMessages.validateTooShort)
+    .max(50, sharedMessages.validateTooLong),
+  description: Yup.string()
+    .max(2000, sharedMessages.validateTooLong),
+  frequency_plan_id: Yup.string()
+    .required(sharedMessages.validateRequired),
+})
+
+@withBreadcrumb('gateways.add', function () {
+  return (
+    <Breadcrumb
+      path="/console/gateways/add"
+      icon="add"
+      content={sharedMessages.add}
+    />
+  )
+})
+@connect(function (state, props) {
+  const userId = userIdSelector(state, props)
+
+  return { userId }
+},
+dispatch => ({
+  createSuccess: () => dispatch(push('/console/gateways')),
+}))
+@bind
+export default class GatewayAdd extends React.Component {
+
+  state = {
+    error: '',
+  }
+
+  async handleSubmit (values, { resetForm }) {
+    const { userId, createSuccess } = this.props
+    await this.setState({ error: '' })
+
+    try {
+      await api.gateway.create(userId, values)
+
+      createSuccess()
+    } catch (error) {
+      resetForm(values)
+
+      await this.setState({ error })
+    }
+  }
+
+  render () {
+    const { error } = this.state
+
+    return (
+      <Container>
+        <Row className={style.wrapper}>
+          <Col sm={12}>
+            <IntlHelmet
+              title={sharedMessages.addGateway}
+            />
+            <Message component="h2" content={sharedMessages.addGateway} />
+          </Col>
+          <Col sm={12} md={8}>
+            <Form
+              submitEnabledWhenInvalid
```

Comment:
While we're at it, can you add this prop also to the application add form to be consistent.

Reply:
no, will do it in a separate pr

Reply:
what consistency? the add application form has 3 fields and only the first one is required.
again:
> it doesnt suffer from the issue we are trying to solve in add device/gateway pages

Reply:
The consistency of the submit button behavior when creating entities. Even better would be to change it throughout all submit buttons, but that's not important enough right now.